src/storage-client/src/source/persist_source.rs

```diff
@@ -36,6 +37,21 @@ use crate::types::sources::SourceData;
 
 pub use mz_persist_client::operators::shard_source::FlowControl;
 use mz_timely_util::buffer::ConsolidateBuffer;
+use mz_timely_util::operator::CollectionExt;
+
+/// Request a particular guarantee about consolidation from the source.
+/// It is legal for the source to satisfy a request for consolidation with an ordinary consolidate
+/// operator, but the hope is for it to be able to do something more clever.
+pub enum Consolidation {
+    /// No guarantees about how the resulting collection's data is consolidated.
+    Whatever,
```

Comment:
Rejected ideas: `None` (to avoid confusion with `Option::None`), `Unconsolidated` (since it doesn't guarantee that it's not consolidated, just permit it.

But do let me know if you like any of the rejected options or have other preferences.

Reply:
I'll throw in `Maybe`